components/projects/Projects.tsx

```diff
@@ -0,0 +1,63 @@
+import { Grid } from "@mantine/core";
+import { useEthers } from "@usedapp/core";
+import { useEffect, useState } from "react";
+import { getManagerContract } from "../../utils/contracts";
+import LoadingCard from "./LoadingCard";
+import ProjectCard from "./ProjectCard";
+
+const Projects = () => {
+  const { library, chainId } = useEthers();
+  const [numProjects, setNumProjects] = useState<number>();
+  const [projects, setProjects] = useState<Array<any>>([]);
+
+  useEffect(() => {
+    let isSubscribed = true;
+
+    const fetchProjects = async () => {
+      setProjects([]);
+      setNumProjects(undefined);
+
+      if (library) {
+        const manager = getManagerContract(library);
+        const len = await manager.getProjectsLength();
+        setNumProjects(len);
+        for (let i = 0; i < len && isSubscribed; i++) {
+          const project = await manager.projects(i);
```

Comment:
We want to load every one of them simultaneously

```suggestion
          const project = manager.projects(i).then(result => setProjects.append(result));
```

Reply:
.append is causing rerender and when `setProjects` is in dependency array of useEffect it causes rerenders over and over again. 
With useState I don't have this issues. I'd like to know why 😄